docs/advanced-features/automatic-static-optimization.md

```diff
@@ -20,6 +20,8 @@ If the above is not the case, Next.js will **statically optimize** your page aut
 
 During prerendering, the router's `query` object will be empty since we do not have `query` information to provide during this phase. After hydration, Next.js will trigger an update to your application to provide the route parameters in the `query` object.
 
+However, with `Rewrites` in your code, a re-render is triggered to provide the parsed parameters. This is because [rewrites](/docs/api-reference/next.config.js/rewrites) can have parameters that we need to provide in the `query` object.
```

Comment:
```suggestion
The cases where the query will be updated after hydration triggering another render for an automatically statically optimized page are:

- The page has query values in the URL.
- The page is a [dynamic-route](/docs/routing/dynamic-routes.md).
- You have [rewrites](/docs/api-reference/next.config.js/rewrites.md) in your `next.config.js` since these can have parameters that may need to be parsed and provided in the `query`. 
```
I think we need a snippet about this for `getStaticProps` pages as well.